editorconfig extension's fields cannot be set due to missing setters

Me again =D. I got the following error with bellow config block

```
editorconfig {
    includes = [ 'myfolder', 'secondfolder' ]
}
```


```
$ gradle editorconfigFormat

FAILURE: Build failed with an exception.

* Where:
Build file '/somefolder/build.gradle' line: 45

* What went wrong:
A problem occurred evaluating root project 'eos-jenkins'.
> Cannot set the value of read-only property 'includes' for object of type org.ec4j.gradle.EditorconfigExtension.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org
```

Indeed it looks like EditorconfigExtension class has no public setter for "includes" and other properties

--- a/README.adoc
+++ b/README.adoc
@@ -61,14 +61,21 @@ check.dependsOn editorconfigCheck
 
 == Configuration
 
-There is `editorconfig` extension object. Its properties are defined in the
+There is `editorconfig` extension object. Its properties are documented in the
 https://github.com/ec4j/editorconfig-gradle-plugin/tree/master/src/main/java/org/ec4j/gradle/EditorconfigExtension.java[EditorconfigExtension]
 class.
 
 [source,gradle]
 ----
 editorconfig {
-   ....
+
+   # All files are included by default. Paths or patterns must be relative to the the project root directory.
+   includes = ['src/**', 'build.gradle']
+
+   # Exlude some files in addition to the default excludes
+   # Paths or patterns must be relative to the the project root directory.
+   excludes = ['derby.log', 'LICENSE']
+
 }
 ----
 

--- a/build.gradle
+++ b/build.gradle
@@ -96,7 +96,7 @@ license {
         "gradlew",
         "gradlew.bat",
         "license-header-template.txt",
-        "src/test/projects/defaults*/**/*.txt"
+        "src/test/projects/**/*.txt"
     ])
     mapping {
         java = 'SLASHSTAR_STYLE'

--- a/src/main/java/org/ec4j/gradle/EditorconfigExtension.java
+++ b/src/main/java/org/ec4j/gradle/EditorconfigExtension.java
@@ -174,4 +174,48 @@ public class EditorconfigExtension {
     public boolean isFailOnNoMatchingProperties() {
         return failOnNoMatchingProperties;
     }
+
+    public void setAddLintersFromClassPath(boolean addLintersFromClassPath) {
+        this.addLintersFromClassPath = addLintersFromClassPath;
+    }
+
+    public void setBackup(boolean backup) {
+        this.backup = backup;
+    }
+
+    public void setBackupSuffix(String backupSuffix) {
+        this.backupSuffix = backupSuffix;
+    }
+
+    public void setEncoding(String encoding) {
+        this.encoding = encoding;
+    }
+
+    public void setExcludeNonSourceFiles(boolean excludeNonSourceFiles) {
+        this.excludeNonSourceFiles = excludeNonSourceFiles;
+    }
+
+    public void setExcludes(List<String> excludes) {
+        this.excludes = excludes;
+    }
+
+    public void setExcludeSubmodules(boolean excludeSubmodules) {
+        this.excludeSubmodules = excludeSubmodules;
+    }
+
+    public void setFailOnFormatViolation(boolean failOnFormatViolation) {
+        this.failOnFormatViolation = failOnFormatViolation;
+    }
+
+    public void setFailOnNoMatchingProperties(boolean failOnNoMatchingProperties) {
+        this.failOnNoMatchingProperties = failOnNoMatchingProperties;
+    }
+
+    public void setIncludes(List<String> includes) {
+        this.includes = includes;
+    }
+
+    public void setLinters(List<LinterConfig> linters) {
+        this.linters = linters;
+    }
 }

--- a/src/test/java/org/ec4j/gradle/EditorconfigGradlePluginITest.java
+++ b/src/test/java/org/ec4j/gradle/EditorconfigGradlePluginITest.java
@@ -49,14 +49,20 @@ public class EditorconfigGradlePluginITest {
         }
     }
 
+    private static void assertNoLogText(String projectName, String logText, String needle) {
+        if (logText.contains(needle)) {
+            Assert.fail("Log text of '" + projectName + "' should not contain '" + needle + "'\n\n" + logText);
+        }
+    }
+
     @BeforeClass
     public static void beforeClass() throws IOException {
         IoTestUtils.deleteDirectory(buildProjectsPath);
     }
 
-    private Path testProjectPath;
+    //private Path testProjectPath;
 
-    private void assertFilesEqual(Path actualBaseDir, Path expectedBaseDir, String relPath) throws IOException {
+    private static void assertFilesEqual(Path actualBaseDir, Path expectedBaseDir, String relPath) throws IOException {
         final String contentActual = new String(Files.readAllBytes(actualBaseDir.resolve(relPath)),
                 StandardCharsets.UTF_8);
         final String contentExpected = new String(Files.readAllBytes(expectedBaseDir.resolve(relPath)),
@@ -67,7 +73,7 @@ public class EditorconfigGradlePluginITest {
     @Test
     public void checkDefaults() throws IOException, InterruptedException {
         final String projectName = "defaults";
-        init(projectName, "checkDefaults");
+        Path testProjectPath = init(projectName, "checkDefaults");
 
         BuildResult result = GradleRunner.create().withProjectDir(testProjectPath.toFile()).withArguments( //
                 EditorconfigCheckTask.NAME //
@@ -78,9 +84,10 @@ public class EditorconfigGradlePluginITest {
                 .withPluginClasspath() //
                 .buildAndFail();
 
-        Assert.assertEquals(result.task(":" + EditorconfigCheckTask.NAME).getOutcome(), TaskOutcome.FAILED);
-
         final String logText = result.getOutput();
+        Files.write(testProjectPath.resolve("log.txt"), logText.getBytes(StandardCharsets.UTF_8));
+
+        Assert.assertEquals(result.task(":" + EditorconfigCheckTask.NAME).getOutcome(), TaskOutcome.FAILED);
 
         assertLogText(projectName, logText,
                 "Processing file '.editorconfig' using linter org.ec4j.maven.linters.TextLinter");
@@ -125,9 +132,58 @@ public class EditorconfigGradlePluginITest {
     }
 
     @Test
+    public void checkExtension() throws IOException, InterruptedException {
+        final String projectName = "extension";
+        final Path testProjectPath = init(projectName, "checkExtension");
+
+        BuildResult result = GradleRunner.create().withProjectDir(testProjectPath.toFile()).withArguments( //
+                EditorconfigCheckTask.NAME //
+                , "--debug" //
+        // , "--stacktrace" //
+        ) //
+                // .withDebug(true)
+                .withPluginClasspath() //
+                .buildAndFail();
+
+        final String logText = result.getOutput();
+        Files.write(testProjectPath.resolve("log.txt"), logText.getBytes(StandardCharsets.UTF_8));
+
+        Assert.assertEquals(result.task(":" + EditorconfigCheckTask.NAME).getOutcome(), TaskOutcome.FAILED);
+
+
+        assertNoLogText(projectName, logText,
+                "Processing file '.editorconfig' using linter org.ec4j.maven.linters.TextLinter");
+        assertNoLogText(projectName, logText,
+                "Processing file 'build.gradle' using linter org.ec4j.maven.linters.TextLinter");
+        assertLogText(projectName, logText,
+                "Processing file 'src/main/java/org/ec4j/maven/it/defaults/App.java' using linter org.ec4j.maven.linters.TextLinter"
+                        .replace('/', File.separatorChar));
+        assertLogText(projectName, logText,
+                "No formatting violations found in file 'src/main/java/org/ec4j/maven/it/defaults/App.java'"
+                        .replace('/', File.separatorChar));
+        assertLogText(projectName, logText,
+                "Processing file 'src/main/resources/trailing-whitespace.txt' using linter org.ec4j.maven.linters.TextLinter"
+                        .replace('/', File.separatorChar));
+        assertLogText(projectName, logText,
+                "src/main/resources/trailing-whitespace.txt@1,7: Delete 2 characters - violates trim_trailing_whitespace = true, reported by org.ec4j.maven.linters.TextLinter"
+                        .replace('/', File.separatorChar));
+        assertNoLogText(projectName, logText,
+                "Processing file 'src/main/resources/indent.xml' using linter org.ec4j.maven.linters.TextLinter"
+                        .replace('/', File.separatorChar));
+        assertNoLogText(projectName, logText,
+                "Processing file 'README.adoc' using linter org.ec4j.maven.linters.TextLinter");
+        assertLogText(projectName, logText, "Checked 2 files");
+        assertLogText(projectName, logText, ":" + EditorconfigCheckTask.NAME + " FAILED");
+        assertLogText(projectName, logText, "There are .editorconfig violations. You may want to run");
+        assertLogText(projectName, logText, "./gradlew editorconfigFormat");
+        assertLogText(projectName, logText, "to fix them automagically.");
+
+    }
+
+    @Test
     public void formatDefaults() throws IOException, InterruptedException {
         final String projectName = "defaults";
-        init(projectName, "formatDefaults");
+        final Path testProjectPath = init(projectName, "formatDefaults");
 
         BuildResult result = GradleRunner.create().withProjectDir(testProjectPath.toFile()).withArguments( //
                 EditorconfigFormatTask.NAME //
@@ -138,9 +194,10 @@ public class EditorconfigGradlePluginITest {
                 .withPluginClasspath() //
                 .build();
 
-        Assert.assertEquals(result.task(":" + EditorconfigFormatTask.NAME).getOutcome(), TaskOutcome.SUCCESS);
-
         final String logText = result.getOutput();
+        Files.write(testProjectPath.resolve("log.txt"), logText.getBytes(StandardCharsets.UTF_8));
+
+        Assert.assertEquals(result.task(":" + EditorconfigFormatTask.NAME).getOutcome(), TaskOutcome.SUCCESS);
 
         assertLogText(projectName, logText,
                 "Processing file '.editorconfig' using linter org.ec4j.maven.linters.TextLinter");
@@ -189,10 +246,11 @@ public class EditorconfigGradlePluginITest {
 
     }
 
-    private void init(String projectName, String testName) throws IOException {
-        this.testProjectPath = buildProjectsPath.resolve(testName);
+    private static Path init(String projectName, String testName) throws IOException {
+        final Path testProjectPath = buildProjectsPath.resolve(testName);
         IoTestUtils.deleteDirectory(testProjectPath);
         IoTestUtils.copyDirectory(srcProjectsPath.resolve(projectName), testProjectPath);
+        return testProjectPath;
     }
 
 }

--- a/src/test/projects/extension/.editorconfig
+++ b/src/test/projects/extension/.editorconfig
@@ -1,0 +1,25 @@
+#
+# Copyright (c) 2017 EditorConfig Maven Plugin
+# project contributors as indicated by the @author tags.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+root = true
+
+[*]
+end_of_line = lf
+trim_trailing_whitespace = true
+insert_final_newline = true
+indent_style = space
+indent_size = 2

--- a/src/test/projects/extension/README.adoc
+++ b/src/test/projects/extension/README.adoc
@@ -1,0 +1,3 @@
+= Defaults integration test
+  
+The previous line contains trailing spaces

--- a/src/test/projects/extension/build.gradle
+++ b/src/test/projects/extension/build.gradle
@@ -1,0 +1,24 @@
+/**
+ * Copyright (c) 2018 EditorConfig Gradle Plugin
+ * project contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+plugins {
+    id 'org.ec4j.editorconfig'
+}
+
+editorconfig {
+    includes = ['src/**']
+    excludes = ['src/main/resources/indent.xml']
+}

--- a/src/test/projects/extension/src/main/java/org/ec4j/maven/it/defaults/App.java
+++ b/src/test/projects/extension/src/main/java/org/ec4j/maven/it/defaults/App.java
@@ -1,0 +1,27 @@
+/*
+ * Copyright (c) 2018 EditorConfig Gradle Plugin
+ * project contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ec4j.maven.it.defaults;
+
+/**
+ * Hello world!
+ *
+ */
+public class App {
+    public static void main(String[] args) {
+        System.out.println("Hello World!");
+    }
+}

--- a/src/test/projects/extension/src/main/resources/indent.xml
+++ b/src/test/projects/extension/src/main/resources/indent.xml
@@ -1,0 +1,25 @@
+<?xml version="1.0"?>
+<!--
+
+    Copyright (c) 2018 EditorConfig Gradle Plugin
+    project contributors as indicated by the @author tags.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+-->
+<!-- license -->
+<root>
+  <parent-1>
+     <text-1>text in text-1</text-1>
+    </parent-1>
+</root>

--- a/src/test/projects/extension/src/main/resources/trailing-whitespace.txt
+++ b/src/test/projects/extension/src/main/resources/trailing-whitespace.txt
@@ -1,0 +1,2 @@
+line 1  
+the previous line ends with whitespace

